src/compiler/checker.ts

```diff
@@ -13848,8 +13848,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             paramMapper = createTypeMapper(right.typeParameters, left.typeParameters);
             // We just use the type parameter defaults from the first signature
         }
+        let flags = (left.flags | right.flags) & (SignatureFlags.PropagatingFlags & ~SignatureFlags.HasRestParameter);
         const declaration = left.declaration;
         const params = combineUnionParameters(left, right, paramMapper);
+        const lastParam = lastOrUndefined(params);

```

Comment:
Perhaps this isn't worth it, but this same code is being done in both `combineSignaturesOfUnionMembers` and `combineSignaturesOfIntersectionMembers`.

- Is it worth making a helper out of this? Probably not, given...
- These two functions in general look really, really similar. Why aren't they the same function with some branching? 🤔

Reply:
ye, that's true - I don't think it's worth the effort unifying them right now though